skbio/metadata/_interval.py

```diff
@@ -559,6 +559,68 @@ def _reverse(self):
         # DONT' forget this!!!
         self._is_stale_tree = True
 
+    @classonlymethod
+    @experimental(as_of="0.5.0-dev")
+    def concat(cls, intvl_metadata):
+        '''Concatenate an iterable of ``IntervalMetadata`` objects.
+
+        It concatenates the multiple ``IntervalMetadata`` objects into
+        one coordinate space. The order of the objects in the input
+        iterable matters. The coordinate of the second
+        ``InterableMetadata`` will be shifted up with the length of
+        the first ``IntervalMetadata`` object.
+
+        This function is useful when you concatenate multiple sequences.
+
+        Parameters
+        ----------
+        intvl_metadata : Iterable (IntervalMetadata)
+            The interval metadata to concatenate.
+
+        Returns
+        -------
+        IntervalMetadata
+            Concatenated interval metadata.
+
+        Examples
+        --------
+        >>> from skbio.metadata import IntervalMetadata
+        >>> im1 = IntervalMetadata(3)
+        >>> _ = im1.add([(0, 2)], [(True, False)], {'gene': 'sagA'})
+        >>> im2 = IntervalMetadata(4)
```

Comment:
If we could have some commentary on what exactly is going on here could help improve readability of this example